src/Lecture1.hs

```diff
@@ -104,7 +117,28 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+
+strSum :: [Char] -> Int
+strSum str1 = sumListAccu (splitToInts str1) 0
+    where
+      splitAccu :: [Char] -> [Char] -> [[Char]] -> [[Char]]
+      splitAccu str strAccu strListAccu
+          | (null str) && (null strAccu) = strListAccu
+          | (null str) = strAccu : strListAccu
+          | (head str == ' ') && (null strAccu) = splitAccu (tail str) [] strListAccu
+          | (head str == ' ') = [strAccu] ++ (splitAccu (tail str) [] strListAccu)
+          | otherwise = splitAccu (tail str) (strAccu ++ [head str]) strListAccu
+    
+      split :: [Char] -> [[Char]]
+      split strList = splitAccu strList [] []
+
+      splitToInts :: [Char] -> [Int]
+      splitToInts list = map (\x -> read x :: Int) (split list)
+
+      sumListAccu:: [Int] -> Int -> Int
+      sumListAccu list accu
+          | null list = accu
+          | otherwise = sumListAccu (tail list) (head list) + accu
```

Comment:
There's a standard function `sum` that finds the sum of all elements in the list. So you don't need to write your own function ✍🏻